src/wallet/wallet.go

```diff
@@ -648,5 +570,93 @@ func cmpUxOutByHash(a, b UxBalance) bool {
 		logger.Panic("Duplicate UxOut when sorting")
 	}
 	return cmp < 0
+}
+
+// ChooseSpends chooses uxouts from a list of uxouts.
+// It first chooses the uxout with the most number of coins that has nonzero coinhours.
+// It then chooses uxouts with zero coinhours, ordered by sortStrategy
+// It then chooses remaining uxouts with nonzero coinhours, ordered by sortStrategy
+func ChooseSpends(uxa []UxBalance, amt Balance, sortStrategy func([]UxBalance)) ([]UxBalance, error) {
+	if amt.Coins == 0 {
+		return nil, errors.New("zero spend amount")
+	}
+
+	if len(uxa) == 0 {
+		return nil, errors.New("no unspents to spend")
+	}
+
+	for _, ux := range uxa {
+		if ux.Coins == 0 {
+			logger.Panic("UxOut coins are 0, can't spend")
```

Comment:
Panic here? this function has return error value

Reply:
This panic was in the old function too, I think it panics because this should never happen and means something is corrupt internally